Library/Homebrew/sandbox.rb

```diff
@@ -63,6 +63,9 @@ def allow_write_temp_and_cache
     allow_write "^/private/var/folders/[^/]+/[^/]+/[C,T]/", :type => :regex
     allow_write_path HOMEBREW_TEMP
     allow_write_path HOMEBREW_CACHE
+    allow_write_path ENV["TMPDIR"] if ENV["TMPDIR"]
+    allow_write_path ENV["TEMP"] if ENV["TEMP"]
+    allow_write_path ENV["TMP"] if ENV["TMP"]
```

Comment:
FYI, `allow_write_path` requires the path to be absolute or it will raise error.